specification/cognitiveservices/AnomalyDetector/univariate/models.cadl

```diff
@@ -0,0 +1,304 @@
+  Auto: "auto",
+  Previous: "previous",
+  Linear: "linear",
+  Fixed: "fixed",
+  Zero: "zero",
+  NotFill: "notFill",
+}
+
+@knownValues(ImputeModeKnownValues)
+model ImputeMode is string;
+
+enum AnomalyDetectorErrorCodesKnownValues {
+  "InvalidCustomInterval",
+  "BadArgument",
+  "InvalidGranularity",
+  "InvalidPeriod",
+  "InvalidModelArgument",
+  "InvalidSeries",
+  "InvalidJsonFormat",
+  "RequiredGranularity",
+  "RequiredSeries",
+  "InvalidImputeMode",
+  "InvalidImputeFixedValue",
+}
+
+@knownValues(AnomalyDetectorErrorCodesKnownValues)
+model AnomalyDetectorErrorCodes is string;
+
+enum TimeGranularity {
+  Yearly: "yearly",
+  Monthly: "monthly",
+  Weekly: "weekly",
+  Daily: "daily",
+  Hourly: "hourly",
+  PerMinute: "minutely",
+  PerSecond: "secondly",
+  Microsecond: "microsecond",
+  None: "none",
+}
+
+@doc("The request of entire or last anomaly detection.")
+model UnivariateDetectionOptions {
+  @doc("""
+Time series data points. Points should be sorted by timestamp in ascending
+order to match the anomaly detection result. If the data is not sorted
+correctly or there is duplicated timestamp, the API will not work. In such
+case, an error message will be returned.
+""")
+  series: TimeSeriesPoint[];
+
+  @doc("""
+Optional argument, can be one of yearly, monthly, weekly, daily, hourly,
+minutely, secondly, microsecond or none. If granularity is not present, it will
+be none by default. If granularity is none, the timestamp property in time
+series point can be absent.
+""")
+  granularity?: TimeGranularity;
+
+  @doc("""
+Custom Interval is used to set non-standard time interval, for example, if the
+series is 5 minutes, request can be set as {\"granularity\":\"minutely\",
+\"customInterval\":5}.
+""")
+  customInterval?: int32;
+
+  @doc("""
+Optional argument, periodic value of a time series. If the value is null or
+does not present, the API will determine the period automatically.
+""")
+  period?: int32;
+
+  @doc("Optional argument, advanced model parameter, max anomaly ratio in a time series.")
+  maxAnomalyRatio?: float32;
+
+  @doc("""
+Optional argument, advanced model parameter, between 0-99, the lower the value
+is, the larger the margin value will be which means less anomalies will be
+accepted.
+""")
+  sensitivity?: int32;
+
+  @doc("""
+Used to specify how to deal with missing values in the input series, it's used
+when granularity is not \"none\".
+""")
+  imputeMode?: ImputeMode;
+
+  @doc("""
+Used to specify the value to fill, it's used when granularity is not \"none\"
+and imputeMode is \"fixed\".
+""")
+  imputeFixedValue?: float32;
+}
+
+@doc("The definition of input timeseries points.")
+model TimeSeriesPoint {
+  @doc("Optional argument, timestamp of a data point (ISO8601 format).")
+  timestamp?: zonedDateTime;
+
+  @doc("The measurement of that point, should be float.")
+  value: float32;
+}
+
+@doc("The response of entire anomaly detection.")
+model UnivariateDetectionResult {
```

Comment:
We recommend "UnivariateDetectionResult => UnivariateEntireDetectionResult" here.